modules/RouteUtils.js

```diff
@@ -96,3 +99,38 @@ export function createRoutes(routes) {
 
   return routes
 }
+
+/**
+ * Checks a Route for the component that SHOULD be rendered against the
+ * component that WILL be rendered. If falsy, this indicates an
+ * unreachable route.
+ */
+function verifyRouteIsReachable(path, component, transitionManager) {
+
+  // strip vars so the path becomes routable
+  const cleanPath = path.replace(':', '')
+
+  transitionManager.match(createLocation(cleanPath), (_, _1, state) => {
+    // if components to render don't include the component we expect to see (from the <Route>), warn
+    warning(state.components.includes(component), `<Route path='${path}' component=${component.name}> is an unreachable route. You may need to reorder your routes or fix a typo.`)
```

Comment:
- Check the `route` in `state.routes`, not the `component` (works better w/e.g. `getComponent` or `getComponents`)
- Don't use `Array.prototype.includes` – even with `babel-runtime`, it doesn't transpile down to ES5, and we don't want to force users to include a polyfill